### ISSUE TITLE
USWDS - Bug: Alert component needs meaningful SR text to convey status

### ISSUE THREAD

**Issue by jeana-adhoc** [uswds/uswds#6539]
### Describe the bug

The icons in the `usa-alert` component do not expose meaningful text equivalents.

Currently, the pseudo-element used for the alert icon (e.g., warning, error, success) is purely decorative from a screen reader’s perspective. The screen reader announces only the visible heading text, such as:

> Your balance may be overdue

Sighted users, however, receive additional meaning through both icon and color treatment. Because the icon conveys status (e.g., warning, error, success), the absence of a text equivalent results in a loss of information for assistive technology users.

### Steps to reproduce the bug

1. [Go to `<usa-alert>` with visible status icon in storybook](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-alert--warning&viewMode=story)
2. Turn on a screen reader (e.g., NVDA or VoiceOver).
3. Move focus to the alert.
4. Listen to what’s announced.

**Actual:**
Only the visible text is announced.

### Expected Behavior

The alert’s type is also announced (e.g., “Warning”, “Success”, “Error”) via sr-text that is in the heading tag.

For example:

> Warning Alert: Your balance may be overdue

The visible alert would look like

> ⚠️ Your balance is overdue

### System setup

- Tested with Chrome, safari, firefox
- Using VO, JAWS, Firefox

### Additional context

The additional option would be to load the alert icon as an image and use alt-text

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by annepetersen** [comment]
Hi Jeana! And thanks for this. Since this is for USWDS, would you mind editing the issue to be specific to USWDS components, rather than `<va-alert>`, for example? 

If not, this comment I'm making here can also suffice to note we're talking about `usa-alert`. Also, here's a more full-context link to [USWDS' Storybook](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-alert-icon-alignment/?path=/story/components-alert--warning)

**Comment by jeana-adhoc** [comment]
Good call out - I've updated. Thank you!

**Comment by annepetersen** [comment]
[Discussing this one further](https://docs.google.com/document/d/1uEsF7KY3WHzDTOCZ4rcdIVSLJYVanmWGc10-_Wj3eqg/edit?tab=t.0#heading=h.ptjhepto4tiq) with @chandracarney and our VA colleagues. @chandracarney's current proposed solution is to outline this in the component guidance for implementers. Something like:

> **Specify the type of alert to the user.** Users should be able to understand the purpose of the alert without relying solely on its color (or the shape of its icon). For example “Success: account created” or “Error: could not log in”

Hopefully making further progress on decision making tomorrow.

**Comment by annepetersen** [comment]
As an update, we're still working through this one, now bringing in a content strategist (thanks @rosamundtgov!) and doing some research to check on how well this approach might work.

**Comment by annepetersen** [comment]
[First draft complete](https://docs.google.com/document/d/1uEsF7KY3WHzDTOCZ4rcdIVSLJYVanmWGc10-_Wj3eqg/edit?tab=t.0#heading=h.x4my8nq3y2f) and ready for internal review. Passing along to folks within TTS and VA for validation and testing before wider release.

**Comment by chandracarney** [comment]
Updated guidance is ready for wider review and a PR will follow any discussion of this update. We plan to add the following guidance to the [alert component](https://designsystem.digital.gov/components/alert/) page:

> **Start the alert’s text with a word describing the alert type.** Specify the type of alert in the first word of its text. Depending on the type of alert component (informative, warning, etc.), that text may be set as heading or body copy. Users should be able to understand the purpose of the alert without relying solely on its color or the shape of its icon. If leading with the alert type makes the alert itself unclear, consider including it in the body text. 


> **Use the appropriate alert type.** 
> - Informative: Non-critical information, such as status. 
>   - For informative alert types, consider using “Announcement” or a more precise term as the first word.
> - Warning: Potentially critical information that might impact the experience or require action. 
> - Success: Information that indicates an action was successful. 
> - Error: Information that indicates an action failed.
> - Emergency: Critical, high-priority information that will impact the experience and might require immediate action.

> **Create short and clear alert titles.** The alert heading should include the alert type and can provide more about the alert, but shouldn’t include more than one line of text. Add any additional alert details in the body text of the alert.

**Comment by chandracarney** [comment]
From my research, I'll pull some key reasoning to support recommending the copy heading should include the alert type as the first word:
- All users benefit from knowing the alert type presented (e.g. “error”), rather than depending on color and icon shape to infer information. 
- If we inject the error type in alt text, screen reader users might be getting duplicative information if a team has already put the alert type in the heading or body copy.
  - Depending on the inherent screen reader behavior, copy choice, and how the component is implemented, this duplicative information may come 3 times: “Alert, error alert, error, your password change was unsuccessful.” 
  - Web Component renderings cause things to break and behave in unexpected ways when injecting alt text to the icon in an alert.
- The incorrect alert type is frequently used (e.g. warning alert used that should be an information alert); this would help design and content staff better choose the alert type.


Looking forward to getting this change in! Please feel free to discuss this guidance update. 🥳

**Comment by caseywatts** [comment]
This looks great! This guidance is going to help a lot 👏

**Comment by chandracarney** [comment]
The new [alert component guidance](https://designsystem.digital.gov/components/alert/) has been added via this PR [#3288](https://github.com/uswds/uswds-site/pull/3288)!
